packages/ui-kit/scripts/parse-design-tokens.mts

```diff
@@ -0,0 +1,342 @@
+}
+
+// Generate design token data
+const generateTokenValue = (key: string, value: any, tokenData: TokenDataProps[]): TokenDataProps => ({
+  cssName: `--propel-${key}`,
+  jsName: kebabCaseToCamelCase(key),
+  kind: value.kind,
+  type: value.type === 'number' ? 'number' : 'string',
+  value: value.isAlias ? parseAlias(value, tokenData) : parseValue(value)
+})
+
+// Generate typography token data
+const generateTypographyValue = (key: string, type: 'number' | 'string', value: string): TokenDataProps =>
+  generateTokenValue(key, { type, value, kind: 'typography' }, [])
+
+// Write design token content to file
+const writeToFileSync = (fileName: string, content: string): void => {
+  const fullPath = path.join('./src/themes/generated/', fileName)
+  const dirName = path.dirname(fullPath)
+
+  // Check if the directory exists. If not, create it
+  if (!fs.existsSync(dirName)) {
+    fs.mkdirSync(dirName, { recursive: true })
+  }
+
+  fs.writeFileSync(fullPath, content)
+}
+
+// Read Figma variables from JSON file
+const getJSONFromFile = async (): Promise<VariablesJSONProps> => {
+  const data = await fsPromises.readFile('./src/themes/variables.json', 'utf-8')
+  const variablesJSON: VariablesJSONProps = JSON.parse(data)
+
+  return variablesJSON
+}
+
+// Get theme tokens
+
+const getThemeTokens = ({
+  name,
+  themes,
+  variables,
+  tokens
+}: {
+  name: string
+  themes?: ThemeProps[]
+  variables: TokenDataProps[]
+  tokens: TokenDataProps[]
+}) =>
+  themes
+    ?.find((theme) => theme.name.toLowerCase() === name)
+    ?.variables.map((variable) =>
+      generateTokenValue(
+        slugifyStr(variable.name.split('/').pop() ?? ''),
+        variable,
+        variable.isAlias ? variables : tokens
+      )
+    )
+
+const main = async () => {
+  const spinner = ora({ text: 'Parsing design variables and tokens...', color: 'yellow' }).start()
+
+  const variablesJSON = await getJSONFromFile()
+
+  if (!variablesJSON) {
+    spinner.fail('Failed to parse variables.json')
+    throw new Error('Failed to parse variables.json')
+  }
+
+  try {
+    const variables: TokenDataProps[] = []
+    const tokens: TokenDataProps[] = []
+
+    // Parse variables
+    variablesJSON.collections
+      .filter(({ name }) => ['_Primitives'].includes(name))
+      .forEach((collection) => {
+        collection.modes[0].variables.forEach((variable) => {
+          variables.push(generateTokenValue(slugifyStr(variable.name), variable, variables))
+        })
+      })
+
+    // Parse tokens
+    variablesJSON.collections
+      .filter(({ name }) => ['2. Radius', '3. Spacing', '4. Widths', 'Effects'].includes(name))
+      .forEach((collection) => {
+        collection.modes[0].variables.forEach((variable) => {
+          tokens.push(generateTokenValue(slugifyStr(variable.name), variable, variables))
+        })
+      })
+
+    type typographyClassProps = {
+      className: string
+      props: { prop: string; value: string }[]
+    }
```

Comment:
We should probably avoid declaring types directly in the middle of the code, can we move this one to the top-level where types are defined? I think we should also set it to Pascal case for consistency with the other types `TypographyClassProps`